File: typegate/tests/random/random.py
from typegraph_next import t, typegraph
from typegraph_next.graph.typegraph import Graph
from typegraph_next.policy import Policy
from typegraph_next.runtimes.random import RandomRuntime

rec = t.struct(
    {
        "uuid": t.uuid(),
        "int": t.integer(),
        "str": t.string(),
        "email": t.email(),
    }
)

users = t.struct(
    {
        "id": t.uuid(),
        "name": t.string(config={"gen": "name"}),
        "age": t.integer(config={"gen": "age", "type": "adult"}),
        "address": t.struct(
            {
                "street": t.string(config={"gen": "address"}),
                "city": t.string(config={"gen": "city"}),
                "postcode": t.string(config={"gen": "postcode"}),
                "country": t.string(config={"gen": "country", "full": True}),
            }
        ),
    }
)

list = t.struct(
    {
        "array_of_array_of_names": t.array(t.array(t.string(config={"gen": "name"}))),
    }
)


@typegraph()
def test_random(g: Graph):
    runtime_1 = RandomRuntime(seed=1)
    runtime_2 = RandomRuntime(seed=1)

    pub = Policy.public()

    g.expose(
        randomRec=runtime_1.gen(rec).with_policy(pub),
        randomUser=runtime_1.gen(users).with_policy(pub),
        randomList=runtime_2.gen(list).with_policy(pub),
    )

File: typegate/tests/random/random.ts
// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
// SPDX-License-Identifier: Elastic-2.0

import { Policy, t, typegraph } from "@typegraph/sdk/index.js";
import { RandomRuntime } from "@typegraph/sdk/runtimes/random.js";

typegraph("random", (g: any) => {
  const random = new RandomRuntime({ seed: 1, reset: "" });
  const pub = Policy.public();

  // test for enum, union, either
  const rgb = t.struct({
    "R": t.float(),
    "G": t.float(),
    "B": t.float(),
  }, { name: "Rgb" });
  const vec = t.struct({ "x": t.float(), "y": t.float(), "z": t.float() }, {
    name: "Vec",
  });

  const rubix_cube = t.struct({ "name": t.string(), "size": t.integer() }, {
    name: "Rubix",
  });
  const toygun = t.struct({ "color": t.string() }, { name: "Toygun" });

  const testStruct = t.struct({
    field: t.union([rgb, vec]),
    toy: t.either([rubix_cube, toygun]),
    educationLevel: t.enum_(["primary", "secondary", "tertiary"]),
    cents: t.float({ enumeration: [0.25, 0.5, 1.0] }),
  });

  g.expose({
    test1: random.gen(
      t.struct({
        email: t.email(),
        country: t.string({}, { config: { gen: "country", full: true } }),
      }),
    ).withPolicy(pub),
    test2: random.gen(
      testStruct,
    ).withPolicy(pub),
  });
});
